manager/volume.go

```diff
@@ -698,3 +711,29 @@ func (m *VolumeManager) UpdateReplicaCount(name string, count int) (v *longhorn.
 	logrus.Debugf("Updated volume %v replica count from %v to %v", v.Name, oldCount, v.Spec.NumberOfReplicas)
 	return v, nil
 }
+
+func (m *VolumeManager) UpdateDataLocality(name string, dataLocalityMode types.DataLocality) (v *longhorn.Volume, err error) {
+	defer func() {
+		err = errors.Wrapf(err, "unable to update data locality for volume %v", name)
+	}()
+
+	if err := types.ValidateDataLocalityMode(dataLocalityMode); err != nil {
+		return nil, err
+	}
+
+	v, err = m.ds.GetVolume(name)
+	if err != nil {
+		return nil, err
+	}
+
+	oldDataLocalityMode := v.Spec.DataLocality
+	v.Spec.DataLocality = dataLocalityMode
+
+	v, err = m.ds.UpdateVolume(v)
```

Comment:
consider making it a noop when the current data locality setting is the same as the requested.

Reply:
Thank you! Fixed